flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/ExecutionGraphToSchedulingTopologyAdapter.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.adapter;
+
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.executiongraph.ExecutionEdge;
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.executiongraph.IntermediateResultPartition;
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingExecutionVertex;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingTopology;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Adapter of {@link ExecutionGraph} to {@link SchedulingTopology}.
+ */
+public class ExecutionGraphToSchedulingTopologyAdapter implements SchedulingTopology {
+
+	private final Map<ExecutionVertexID, DefaultExecutionVertex> executionVerticesById;
+
+	private final List<SchedulingExecutionVertex> executionVerticesList;
+
+	private final Map<IntermediateResultPartitionID, ? extends SchedulingResultPartition> resultPartitionsById;
+
+	public ExecutionGraphToSchedulingTopologyAdapter(ExecutionGraph graph) {
+		checkNotNull(graph, "execution graph can not be null");
+
+		final int totalVertexCnt = graph.getTotalNumberOfVertices();
+		int totalPartitionCnt = 0;
+		for (ExecutionJobVertex jobVertex : graph.getAllVertices().values()) {
+			totalPartitionCnt += jobVertex.getProducedDataSets().length * jobVertex.getParallelism();
+		}
+
+		this.executionVerticesById = new HashMap<>(totalVertexCnt);
```

Comment:
I'd suggest to use the constructor with no arguments because:
- It is easier to read
- The default load factor is 0.75, i.e., after inserting approx. 0.75 * totalVertexCnt, elements will be re-hashed anyways
- Using `ceil(totalVertexCnt/0.75)` would be harder to understand
- In this code path there won't be a noticeable performance hit due to re-hashing

Also see https://stackoverflow.com/questions/564899/choosing-the-initial-capacity-of-a-hashset-with-an-expected-number-of-unique-val

Reply:
OK, the reason is reasonable.